common/beerocks/bwl/shared/nl80211_socket.cpp

```diff
@@ -0,0 +1,85 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2020 MaxLinear
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "nl80211_socket.h"
+
+#include <netlink/genl/ctrl.h>
+#include <netlink/genl/family.h>
+#include <netlink/genl/genl.h>
+
+#include <easylogging++.h>
+
+/**
+ * Receive and transmit socket buffer size in bytes.
```

Comment:
Honestly, I don't know. I've been investigating and found that the call to `nl_socket_set_buffer_size` ends up calling `setsockopt(SO_SNDBUF)` and `setsockopt(SO_RCVBUF)`. Some people report that they were losing events with default values and had to increase the receive buffer size to fix it. Some tutorials I've found also change default sizes but don't explain why.

I borrowed this code from `base_wlan_hal_nl80211::send_nl80211_msg` and thought that who initially wrote that code had considered it was necessary to call `nl_socket_set_buffer_size` so I kept it. I didn't question myself why it was actually needed nor if values used were the most appropriate.

While investigating, I've realized that the call to `nl_socket_set_buffer_size` requires the socket to be connected or otherwise it fails with NLE_BAD_SOCK. Thus I've removed the call from the constructor and put it after the socket connection has been made.

` base_wlan_hal_nl80211` is also calling `nl_socket_set_buffer_size` but before connecting the socket so I guess it must be failing too (no error code is being checked after the call) and buffer size is not being changed. Thus it looks like, for the commands being executed up to now in BWL, it's not necessary to increase buffer size.